### DISCUSSION TITLE
How can I get mentions of my user and replied tweet's video?

### DISCUSSION THREAD

**Discussion by ErcinDedeoglu** [Xwilarg/TwitterSharp#53] [General]
1 - I want to get tweets that mention my username.
2 - Then I want to get that tweet's parent tweet (replied tweet)
3 - Then I want to get the parent tweet's content, and it's media (mp4, etc) if exists.

Is that possible with this library?

**Comment by marcogruhl** [comment]
Hello @ErcinDedeoglu 

yes, it is possible.

- request `_client.GetRecentTweets(expression, searchOptions);` 
where the expression is like `Expression.Mention("yourNameWithoutAt").And(Expression.IsReply());`
and searchOptions is like `new TweetSearchOptions { TweetOptions = new[] { TweetOption.Referenced_Tweets }};`
- than you have to iterate through the results and request the tweets like `_client.GetTweetAsync(tweet.ReferencedTweets[0].Id, searchOptions);`
where searchOptions is like `new TweetSearchOptions { TweetOptions = new[] { TweetOption.Attachments }};`
